lib/api/controllers/documentController.js

```diff
@@ -1,32 +1,77 @@
 'use strict';
 
 var
+  BadRequestError = require('kuzzle-common-objects').errors.BadRequestError,
   assertHasBody = require('./util/requestAssertions').assertHasBody,
   assertHasId = require('./util/requestAssertions').assertHasId,
-  assertHasIndex = require('./util/requestAssertions').assertHasIndex,
-  assertHasIndexAndCollection = require('./util/requestAssertions').assertHasIndexAndCollection,
   assertBodyHasAttribute = require('./util/requestAssertions').assertBodyHasAttribute,
-  BadRequestError = require('kuzzle-common-objects').errors.BadRequestError;
+  assertHasIndexAndCollection = require('./util/requestAssertions').assertHasIndexAndCollection;
 
 /**
  * @param {Kuzzle} kuzzle
  * @constructor
  */
-function WriteController (kuzzle) {
+function DocumentController (kuzzle) {
   var engine = kuzzle.services.list.storageEngine;
 
+
+  /**
+   * @param {Request} request
+   * @returns {Promise<Object>}
+   */
+  this.search = function documentSearch (request) {
+    assertHasIndexAndCollection(request, 'search');
```

Comment:
as you changed "action" parameter above in other controllers, should we also replace `'search'` by `'document::search'` ?